api/rest.go

```diff
@@ -3436,6 +3482,37 @@ func handleCreateSource(exec boil.Executor, r CreateSourceRequest) (*Source, *Ht
 			return nil, NewInternalError(err)
 		}
 	}
+	// create CU type source
+
+	cuUid := s.UID
+	hasCU, err := models.ContentUnits(exec, qm.Where("uid = ?", cuUid)).Exists()
+	if err != nil {
+		return nil, NewInternalError(err)
+	}
+	if hasCU {
+		cuUid, err = GetFreeUID(exec, new(ContentUnitUIDChecker))
+		if err != nil {
+			return nil, NewInternalError(err)
+		}
+	}
+
+	props := make(map[string]interface{})
+	props["source_id"] = s.UID
+	p, _ := json.Marshal(props)
+
+	cu := &models.ContentUnit{
+		UID:        s.UID,
+		TypeID:     common.CONTENT_TYPE_REGISTRY.ByName[common.CT_SOURCE].ID,
+		Secure:     common.SEC_PUBLIC,
+		Published:  true,
+		Properties: null.JSONFrom(p),
+		CreatedAt:  time.Now(),
+	}
+
+	err = s.Source.AddContentUnits(exec, true, cu)
```

Comment:
This association is wrong. We should not link the unit to the source this way. The link is via the "source_id" property, nothing more.

This will show up in filters (UI) as a unit (lesson, program, etc...) which is about the source. Not the source itself.